assets/js/src/form_editor/store/reducers/history_record.ts

```diff
@@ -0,0 +1,83 @@
+import { isEqual } from 'lodash';
+
+export const createHistoryRecord = (state) => {
+  let editorHistory: any[] = state.editorHistory;
+  let editorHistoryOffset = state.editorHistoryOffset;
+  const lastHistoryRecord = editorHistory[editorHistory.length - 1];
+
+  let newHistoryRecord;
+  const time = Date.now();
+
+  if (
+    lastHistoryRecord !== undefined
+    && (time - lastHistoryRecord.time) < 1000
+    && editorHistory.length > 1
+  ) {
```

Comment:
I think this function need either split into a couple of smaller ones and/or some comments why do we need all of the logic here.